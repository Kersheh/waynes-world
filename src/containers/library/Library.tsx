import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from 'types';
import { getLibraryAction } from './libraryActions';

import styles from './Library.module.scss';

const LibraryContainer = () => {
  const dispatch = useDispatch();
  const { albums } = useSelector((state: RootState) => state.library);

  useEffect(() => {
    dispatch(getLibraryAction());
  }, [dispatch]);

  return (
    <div className={styles.library}>
      Library
      <pre>{JSON.stringify(albums, null, 2)}</pre>
    </div>
  );
};

export default LibraryContainer;
