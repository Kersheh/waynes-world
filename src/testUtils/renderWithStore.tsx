import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
