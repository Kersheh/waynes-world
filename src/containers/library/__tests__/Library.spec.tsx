import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import Library from '../Library';

describe('<Library/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(<Library />);

    expect(container).toMatchSnapshot();
  });
});
