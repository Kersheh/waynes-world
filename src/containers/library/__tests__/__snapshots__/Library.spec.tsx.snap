// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Library/> component should render 1`] = `
<div>
  <div
    class="library"
  >
    Library
    <pre>
      []
    </pre>
  </div>
</div>
`;
